reframe/core/schedulers/lsf.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+#
+# LSF backend
+#
+# - Initial version submitted by Ryan Goodner, UNM (based on PBS backend)
+#
+
+import functools
+import re
+import time
+
+import reframe.core.runtime as rt
+import reframe.utility.osext as osext
+from reframe.core.backends import register_scheduler
+from reframe.core.exceptions import JobSchedulerError
+from reframe.core.schedulers.pbs import PbsJobScheduler
+
+_run_strict = functools.partial(osext.run_command, check=True)
+
+
+@register_scheduler('lsf')
+class LsfJobScheduler(PbsJobScheduler):
+    def __init__(self):
+        self._prefix = '#BSUB'
+        self._submit_timeout = rt.runtime().get_option(
+            f'schedulers/@{self.registered_name}/job_submit_timeout'
+        )
+
+    def emit_preamble(self, job):
+        num_tasks = job.num_tasks or 1
+        num_tasks_per_node = job.num_tasks_per_node or 1
+        num_nodes = int(num_tasks // num_tasks_per_node)
+
+        preamble = [
+            self._format_option(f'-J {job.name}'),
+            self._format_option(f'-o {job.stdout}'),
+            self._format_option(f'-e {job.stderr}'),
+            self._format_option(f'-nnodes {num_nodes}')
+        ]
+
+        # add job time limit in minutes
+        if job.time_limit is not None:
+            preamble.append(
+                self._format_option(f'-W {int(job.time_limit // 60)}')
+            )
+
+        # emit the rest of the options
+        options = job.options + job.cli_options
+        for opt in options:
+            if opt.startswith('#'):
+                preamble.append(opt)
+            else:
+                preamble.append(self._format_option(opt))
+
+        # change to working dir with cd
+        preamble.append(f'cd {job.workdir}')
+
+        return preamble
+
+    def submit(self, job):
+        cmd = f'bsub {job.script_filename}'
+        completed = _run_strict(cmd, timeout=self._submit_timeout)
+        jobid_match = re.search(r'^Job <(?P<jobid>\S+)> is submitted',
+                                completed.stdout)
+        if not jobid_match:
+            raise JobSchedulerError('could not retrieve the job id '
+                                    'of the submitted job')
+
+        job._jobid = jobid_match.group('jobid')
+        job._submit_time = time.time()
+
+    def poll(self, *jobs):
+        if jobs:
+            # filter out non-jobs
+            jobs = [job for job in jobs if job is not None]
+
+        if not jobs:
+            return
+
+        completed = osext.run_command(
+            f'bjobs -noheader {" ".join(job.jobid for job in jobs)}')
+        if completed.returncode != 0:
+            raise JobSchedulerError(
+                f'bjobs failed with exit code {completed.returncode} '
+                f'(standard error follows):\n{completed.stderr}'
```

Comment:
This should be enough:

```suggestion
        completed = _run_strict(
            f'bjobs -noheader {" ".join(job.jobid for job in jobs)}'
        )
```

Reply:
done